### PR TITLE
feat: Integrate new EvmEnvProvider in RPC

### LINKED ISSUE #1546: Integrate new EvmEnvProvider in RPC
### Describe the feature

ref #1525

new trait for fetching Envs needs to be added to `EthApi` and integrated into rpc builder pipeline

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by chirag-bgh**
Closes #1546

### CODE REVIEW

**Comment by chirag-bgh** [comment]
@mattsse is this correct or what I should've done was just adding ```EvmEnvProvider``` to ``` Client: BlockProvider + StateProviderFactory + static,``` everywhere?

File: bin/reth/src/args/rpc_server_args.rs [hunk 1]
```diff
@@ -103,46 +103,60 @@ impl RpcServerArgs {
     }
 
     /// Convenience function for starting a rpc server with configs which extracted from cli args.
-    pub(crate) async fn start_rpc_server<Client, Pool, Network>(
+    pub(crate) async fn start_rpc_server<Client, Pool, Network, Provider>(
```
Comment (mattsse) [inline bin/reth/src/args/rpc_server_args.rs:106]: `Provider` is part of the `Client` so this is not required.

File: bin/reth/src/args/rpc_server_args.rs [hunk 2]
```diff
@@ -103,46 +103,60 @@ impl RpcServerArgs {
...
         &self,
         client: Client,
         pool: Pool,
         network: Network,
+        evm_env_provider: Provider,
     ) -> Result<RpcServerHandle, RpcError>
     where
         Client: BlockProvider + HeaderProvider + StateProviderFactory + Clone + 'static,
```
Comment (mattsse) [inline bin/reth/src/args/rpc_server_args.rs:114]: EvmEnvProvider needs to go here instead as `+ EvmEnvProvider`

**Review by mattsse — CHANGES_REQUESTED** [review]
extend the existing Client trait bounds instead of adding a new argument

**Review by mattsse — APPROVED** [review]
lgtm

**Pull request merged**